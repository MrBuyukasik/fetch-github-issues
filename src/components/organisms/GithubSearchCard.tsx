import React from 'react';
import {StyleProp, TextStyle} from 'react-native';
import styled from 'styled-components/native';
import Button from '../molecules/Button';
import TextInput from '../molecules/TextInput';

const StyledBottom = styled(Button)`
  margin-top: 20px;
`;

interface Props {
  style?: StyleProp<TextStyle> | undefined;
  onClick: () => void;
  organization?: any;
  repository?: any;
  author?: any;
}

const GithubSearchCard = (props: Props) => {
  const {organization, repository, onClick} = props;
  const ORGANIZATION_LABEL = 'Enter Organization name';
  const REPOSITORY_LABEL = 'Enter Repository name';
  const SEARCH_LABEL = 'Search Issues';

  return (
    <>
      <TextInput
        testID={ORGANIZATION_LABEL}
        label={ORGANIZATION_LABEL}
        onUpdate={organization}
      />

      <TextInput
        testID={REPOSITORY_LABEL}
        label={REPOSITORY_LABEL}
        onUpdate={repository}
      />

      <StyledBottom
        testID={SEARCH_LABEL}
        label={SEARCH_LABEL}
        onClick={onClick}
      />
    </>
  );
};

export default GithubSearchCard;
