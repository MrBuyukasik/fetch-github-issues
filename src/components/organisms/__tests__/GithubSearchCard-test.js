import React from 'react';
import renderer from 'react-test-renderer';
import GithubSearchCard from '../GithubSearchCard';
fetch = jest.fn(() => Promise.resolve());

describe('GithubSearchCard Testing', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<GithubSearchCard />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
