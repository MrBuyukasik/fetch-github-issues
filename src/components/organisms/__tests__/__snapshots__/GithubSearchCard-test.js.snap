// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GithubSearchCard Testing renders correctly 1`] = `
Array [
  <View
    style={
      Array [
        Object {
          "marginBottom": 16,
          "marginLeft": 16,
          "marginRight": 16,
          "marginTop": 16,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 16,
            "width": "100%",
          },
          Object {
            "marginBottom": 6,
            "marginLeft": 6,
            "marginRight": 6,
            "marginTop": 6,
          },
        ]
      }
      testID="Enter Organization name"
    >
      Enter Organization name
    </Text>
    <TextInput
      onChangeText={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "#FFFFFF",
            "borderColor": "#000000",
            "borderRadius": 10,
            "borderStyle": "solid",
            "borderWidth": 1,
            "fontSize": 18,
            "height": 50,
            "marginBottom": 2,
            "marginLeft": 2,
            "marginRight": 2,
            "marginTop": 2,
            "paddingBottom": 10,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 10,
            "width": "100%",
          },
        ]
      }
    />
  </View>,
  <View
    style={
      Array [
        Object {
          "marginBottom": 16,
          "marginLeft": 16,
          "marginRight": 16,
          "marginTop": 16,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 16,
            "width": "100%",
          },
          Object {
            "marginBottom": 6,
            "marginLeft": 6,
            "marginRight": 6,
            "marginTop": 6,
          },
        ]
      }
      testID="Enter Repository name"
    >
      Enter Repository name
    </Text>
    <TextInput
      onChangeText={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "#FFFFFF",
            "borderColor": "#000000",
            "borderRadius": 10,
            "borderStyle": "solid",
            "borderWidth": 1,
            "fontSize": 18,
            "height": 50,
            "marginBottom": 2,
            "marginLeft": 2,
            "marginRight": 2,
            "marginTop": 2,
            "paddingBottom": 10,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 10,
            "width": "100%",
          },
        ]
      }
    />
  </View>,
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#3DBDFF",
        "borderRadius": 12,
        "height": 50,
        "marginTop": 20,
        "opacity": 1,
        "width": "50%",
      }
    }
    testID="Search Issues"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 16,
              "width": "100%",
            },
            Object {
              "textAlign": "center",
            },
          ]
        }
      >
        Search Issues
      </Text>
    </View>
  </View>,
]
`;
